src/LandingPage.tsx

```diff
@@ -65,31 +67,51 @@ const SignIn = styled.div`
 `;
 
 const LandingPage: React.FC = () => {
+  const auth = useContext(AuthContext);
+
   function loginSuccessHandler(
-    response: GoogleLoginResponse | GoogleLoginResponseOffline
+    googleResponse: GoogleLoginResponse | GoogleLoginResponseOffline
   ) {
-    console.log(response);
-    let onlineResponse = response as GoogleLoginResponse;
+    let onlineResponse = googleResponse as GoogleLoginResponse;
     if (!onlineResponse) {
       // Not supported yet
       return;
     }
-    // TODO: Send onlineResponse.tokenId to the backend to validate the token, get the user object, and store that in localStorage
-    localStorage.setItem("token", onlineResponse.tokenId);
-    localStorage.setItem("user", onlineResponse.profileObj.name);
+
+    const options = {
+      credentials: "include" as const,
+      method: "POST",
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+      body: `idtoken=${onlineResponse.tokenId}`,
+    };
+    return fetch(`http://paralibrary.digital/api/login`, options)
+      .then(async (response) => {
+        if (response.status === 200) {
+          let json = await response.json();
+          auth.login({ authenticated: true, userId: json.userId });
+        }
+      })
+      .catch((error) => {
+        console.error(error);
+      });
   }
 
   const loginFailureHandler = (error: GoogleLoginResponse) => {
     console.log(error);
   };
+
+  if (auth.credential.authenticated) {
+    return <Redirect to="/library" />;
```

Comment:
Just curious, when you do this, does the nav sidebar show up? I don't really know how redirects work.

Reply:
Yeah! It takes you to the normal library page, as expected. I'm not 100% sure on it either, but I think it calls back to the main App and that uses the router to take you to the correct page